developer/guides/components/demo-component.mdx

```diff
@@ -0,0 +1,559 @@
+
+At the end, you'll get a feel for how to build [built-in components](/developer/guides/components/built-in-components) for Makeswift and make them editable using controls.
+
+Here’s a sneak peak of the final component:
+
+![Final accordion component](/images/developer/guides/components/demo-component/accordion-component.png)
+
+## Prerequisites
+
+1. **Create a new Makeswift site:** We’re going to start with the Visual Builder and choose a blank template. Create a new blank page in the Navigation Sidebar.
+
+2. **Create a Next.js app:** Follow the [Automatic Installation](https://nextjs.org/docs/app/getting-started/installation#automatic-installation) instructions and make sure to include Typescript and Tailwind CSS. Then, follow the [App Router Installation](/developer/app-router/installation) guide to connect your Next.js app to your Makeswift site.
+
+## Build the React component
+
+First, we’ll build a React component that we can then integrate into Makeswift.
+
+To follow along, create the folder structure below.
+
+```
+src/
+├── components/
+    └── Accordions/
+        ├── index.tsx
+        └── Accordions.makeswift.tsx
+```
+
+Let’s start by creating a component with a basic `div` element inside of `index.tsx`.
+
+```tsx src/components/Accordions/index.tsx
+export function Accordions() {
+  return <div>Hello World</div>;
+}
+```
+
+### Register the component with Makeswift
+
+For components to become available to use inside of Makeswift, they need to be registered. We’ll do this in the `Accordions.makeswift.tsx` file.
+
+Here, we’ll reference the `runtime` that was created during the installation step to call its `registerComponent` function and pass the React component we just created as the first prop. Then, we’ll need to pass a configuration object with the following properties.
+
+- `type`: serves as an identifier for the component
+- `label`: shows up as its name in Makeswift
+- `props`: uses Makeswift [controls](/developer/reference/controls/checkbox) to pass props to the React component. We'll leave the `props` empty for now
+
+```tsx src/components/Accordions/Accordion.makeswift.tsx
+import { runtime } from "src/makeswift/runtime";
+
+import { Accordions } from "./";
+
+runtime.registerComponent(Accordions, {
+  type: "accordions",
+  label: "Accordions",
+  props: {},
+});
+```
+
+Learn more about [registering a component](/developer/guides/how-to/built-in-components#registering-the-component) in Makeswift.
+
+### Import the registered component
+
+Registered components need to be imported to show up in the Component Tray.
+
+We can copy the following line into our `src/makeswift/components.tsx` file, a file created during the installation step.
+
+```tsx src/makeswift/components.tsx
+import "src/components/Accordions/Accordions.makeswift";
+```
+
+Now, in Makeswift, we should see the `Accordions` component in the Component Toolbar, which we can then drag into the Canvas.
```

Comment:
Maybe link Component Toolbar to https://docs.makeswift.com/product/builder-basics#component-toolbar?